spec/models/course/assessment/question/programming_spec.rb

```diff
@@ -24,18 +24,61 @@
   let(:instance) { create(:instance) }
   with_tenant(:instance) do
     describe 'callbacks' do
-      subject { build(:course_assessment_question_programming) }
+      let(:question_attributes) { [] }
+      subject { build(:course_assessment_question_programming, *question_attributes) }
+
+      describe 'before_save' do
+        with_active_job_queue_adapter(:test) do
+          before { subject.send(:clear_attribute_changes, :attachment) }
+
+          context 'when a package is removed' do
+            let(:question_attributes) do
+              [{
+                 template_file_count: 1,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.